File: Source/typescript/features/IFeaturesProvider.ts
// Copyright (c) Dolittle. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

import { Features } from './Features';

import { Observable } from 'rxjs';

/**
 * Defines a system that can provide {@link Features}
 */
export abstract class IFeaturesProvider {

    /**
     * Features {@link Observable} providing {@link Features}.
     */
    abstract readonly features: Observable<Features>;
}

File: Source/DotNET/Backend/Features/IFeaturesProvider.cs
// Copyright (c) Dolittle. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

using System;
using System.Reactive.Subjects;

namespace Dolittle.Vanir.Backend.Features
{
    /// <summary>
    /// Defines a system that can provide <see cref="Features"/>.
    /// </summary>
    public interface IFeaturesProvider
    {
        /// <summary>
        /// Features <see cref="Subject{T}"/> providing <see cref="Features"/>.
        /// </summary>
        IObservable<Features> Features { get; }
    }
}
